implementations/elixir/ockam/ockam/lib/ockam/secure_channel/encrypted_transport_protocol/aead_aes_gcm.ex

```diff
@@ -70,7 +70,10 @@ defmodule Ockam.SecureChannel.EncryptedTransportProtocol.AeadAesGcm do
 
     with {:ok, decrypted, data} <- decrypt(encrypted, data),
          {:ok, decoded} <- Wire.decode(decrypted) do
-      message = Message.trace(decoded, data.plaintext_address)
+      message =
+        decoded
+        |> Message.trace(data.plaintext_address)
+        |> Message.put_local_metadata(:transport, :secure_channel)
```

Comment:
this looks a bit different than the other  `:transport`  values  (in the sense that `:tcp` and `:udp` do seems similar,  but `:secure_channel` is something that happens after that?). 
Would it be better if instead of overriding the `:transport`  it set another attribute like `:identity`? .  And the authorization that enforces messages  come from a secure channel then check the presence of this attribute   (latter can check for specific identities as well)

Reply:
This field is used to specify the place where message was unwrapped. Because wrapping a message destroys the local metadata, every time message is wrapped and unwrapped this field should be added to specify the entry point to the local scope.
For example in pipe wrappers: https://github.com/build-trust/ockam/pull/2877/files#diff-9fa81d2fe4269b5dcba95fcff4b59b4323e92e9e55022d53fbf6efba839449c4R25

Maybe we could use a different name for that or even have a special value for "local" messages (instead of no value). I was thinking of maybe using `:channel` or something like that. WDYT?

Reply:
:ok_hand:  fair enough
for me `transport` is a bit unclear as  a secure channel (I _think_ ?) is ultimately a worker, while a tcp `transport` is not?   But also I'm not too familiar with the nomenclature,  I think we can get back to this latter if needed for clarity